apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.html.ts

```diff
@@ -225,3 +255,118 @@ function PopoverEndDate() {
     </p>
   `.toString();
 }
+
+function CreateCourseInstanceModal({
+  courseShortName,
+  csrfToken,
+  initialStartDateFormatted,
+  initialEndDateFormatted,
+  timezone,
+}: {
+  courseShortName: string;
+  csrfToken: string;
+  initialStartDateFormatted: string;
+  initialEndDateFormatted: string;
+  timezone: string;
+}) {
+  return Modal({
+    id: 'createCourseInstanceModal',
+    title: 'Create course instance',
+    formMethod: 'POST',
+    body: html`
+      <div class="form-group">
+        <label for="long_name">Long Name</label>
+        <input
+          type="text"
+          class="form-control"
+          id="long_name"
+          name="long_name"
+          required
+          aria-describedby="long_name_help"
+        />
+        <small id="long_name_help" class="form-text text-muted">
+          The full course instance name, such as "Fall 2019". Users see it joined to the course
+          name, e.g. "${courseShortName} Fall 2019".
+        </small>
+      </div>
+      <div class="form-group">
+        <label for="short_name">Course Instance Identifier (CIID)</label>
```

Comment:
@mwest1066 we don't really use the term "CIID" elsewhere the way we do e.g. "QID" (though we do use it on the course instance settings page). We call this "short name" in the database. Any thoughts on what we should do here?

Reply:
I think we should avoid the term CIID because we don't use it elsewhere. "Short name" or "abbreviation"? Maybe we should change the explanation to something like:
> A short name (less than 8 letters) such as "Fa25" or "W25b". This is used in menus and headers where a short description is required.

Reply:
I think "short name" might be best.

We don't enforce any length restrictions, so I'm inclined not to mention anything about length here. We have lots of people that are using longer names in practice. Some people are integrating section/instructor names in here, which IMO is totally valid.

Reply:
Just raising #9908. Not necessarily saying this needs to be addressed here, but something to keep in mind. The issue raises the option of removing that requirement, but keeps open the option of reviewing them to ensure they are reasonable (e.g., no `//`, no paths including `.git`, possibly avoid non-ASCII characters), and in such cases this may be a valid option to consider here, since the CIID is also a directory name.

Reply:
Good point, I think restricting characters/format would be a good idea.

Reply:
@mwest1066 this PR does currently restrict the characters, see the `pattern` attribute on the input. I think Jonatan was actually suggesting _loosening_ the restriction. IMO, we should save that for when we can more comprehensively address #9908 if we choose to.

Reply:
On that note - @miguelaenlle IIRC a previous iteration called out that this can only use letters/numbers/dashes/underscores without spaces. I think that was nice, as otherwise there's no way for users to discover that except through trial and error.

Reply:
<img width="994" alt="image" src="https://github.com/user-attachments/assets/2a68a853-c28a-4d1d-8602-68c8f5268c5c" />

We do use CIID. I was going to suggest changing the name as well, but stopped myself once I noticed that. maybe update this other page too?

Reply:
I think we should change that page too. I think `QID` is useful terminology because we actually use them to refer to questions, but `AID`/`QIID` shouldn't be used (we should use "short name" instead). I think "Name" and "Short name" should be our standard terminology. (that is, "Name" rather than "Long name")

Reply:
I guess I think that "title" sounds a bit strange for a course instance, while it's reasonable for questions and assessments. I don't have a great argument about why this is, however 😅. And I also agree for consistency "title" for everything is probably the best solution.

And I agree this is way out of scope for this PR!

Reply:
Yeah, agreed that "title" for a course instance does sounds weird. Maybe scratch that and we just don't worry about that particular inconsistency.

Think we should change to just "name" in this PR? Or save this for another one where we can make this all consistent?

Reply:
> On that note - @miguelaenlle IIRC a previous iteration called out that this can only use letters/numbers/dashes/underscores without spaces. I think that was nice, as otherwise there's no way for users to discover that except through trial and error.

I updated the Short name field description to also have instructions on the format to use:

<img width="796" alt="image" src="https://github.com/user-attachments/assets/a021d355-787a-400c-9dde-3063ab22ec30" />

Reply:
Talked with @miguelaenlle about this offline, we're going to save any larger naming changes for another PR where we can make those changes holistically.